platforms/common/px4_log.cpp

```diff
@@ -138,12 +138,12 @@ __EXPORT void px4_log_modulename(int level, const char *module_name, const char
 		if (use_color) {
 			// alway reset color
 			const ssize_t sz = math::min(pos, max_length - (ssize_t)strlen(PX4_ANSI_COLOR_RESET) - (ssize_t)1);
-			pos += sprintf(buf + sz, "%s\n", PX4_ANSI_COLOR_RESET);
+			pos += snprintf(buf + sz, math::max(max_length - sz, (ssize_t)0), "%s\n", PX4_ANSI_COLOR_RESET);
 
 		} else
 #endif // PX4_LOG_COLORIZED_OUTPUT
 		{
-			pos += sprintf(buf + math::min(pos, max_length - (ssize_t)1), "\n");
+			pos += snprintf(buf + math::min(pos, max_length - (ssize_t)1), 1, "\n");
```

Comment:
```suggestion
			pos += snprintf(buf + math::min(pos, max_length - (ssize_t)1), 2, "\n");
```

Looks like this one causes dropping newlines in CI:
```
[  23.869|gzserver  ] [  23.871|px4       ] INFO  [simulator_mavlink] Waiting for simulator to accept connection on TCP port 4560INFO  [simulator_mavlink] Simulator connected on TCP port 4560.INFO  [lockstep_scheduler] setting initial absolute time to 1616000 usINFO  [commander] LED: open /dev/led0 failed (22)WARN  [health_and_arming_checks] Preflight Fail: Airspeed selector module downWARN  [health_and_arming_checks] Preflight Fail: ekf2 missing dataINFO  [tone_alarm] home setINFO  [tone_alarm] notify negativeINFO  [commander] Ready for takeoff!
```

Reply:
ouch, thanks!